### DISCUSSION TITLE
关于禁用拖放

### DISCUSSION THREAD

**Discussion by ifpelset** [CefView/QCefView#325] [Q&A] [answered]
是否有办法临时禁用窗口渲染模式下的拖放操作？调试发现拖放是chromium内部启动的，OnDragEnter返回值并不能禁用chromium拖放。

**Comment by L-Super** [comment, marked as answer]
NCW下可以通过以下接口设置：

https://github.com/CefView/QCefView/blob/37f6ae587b7be2bd2938c70bd6aecf3ef3dfb9e7/include/QCefView.h#L264

**Reply by ifpelset** [reply]
OnDragEnter就是调用的该getter，但是无奈chromium依然会调用系统的DoDragDrop。

**Reply by ifpelset** [reply]
我这边的使用场景如下：js端通过ondragstart回调调用c++端并传递一个文件路径，c++端收到调用后通过QDrag exec启动一个系统级别的拖放操作。会出现崩溃在chromium中启用的拖放操作。因OSR模式不会有拖放操作实现，该流程正常。故想看下是否有办法可以临时禁用窗口模式下的拖放。

**Reply by ifpelset** [reply]
CEF论坛给的方案是使用RevokeDragDrop(hWndHostChild); 但我仅仅想要临时禁用。

**Reply by L-Super** [reply]
不太理解你的意思，通常情况下，将`setEnableDragAndDrop`设置为false即禁用了拖放操作

**Reply by ifpelset** [reply]
意思就是chromium调用流程是DoDragDrop->OnDragEnter->getEnableDragAndDrop。我想达到的效果是如何可以实现让chromium不调用DoDragDrop。

**Reply by L-Super** [reply]
那是属于chromium内部的操作逻辑了，如果要从根源禁止调用的话，得修改chromium源码才能实现了。
